chips/stm32f429zi/src/ethernet/mod.rs

```diff
@@ -0,0 +1,2328 @@
+use kernel::utilities::registers::interfaces::{ReadWriteable, Readable, Writeable};
+use kernel::utilities::registers::{register_bitfields, register_structs, ReadOnly, ReadWrite};
+use kernel::utilities::StaticRef;
+use kernel::ErrorCode;
+
+use crate::rcc;
+use crate::rcc::PeripheralClock;
+use crate::rcc::PeripheralClockType;
+
+mod transmit_descriptor;
+use crate::ethernet::transmit_descriptor::TransmitDescriptor;
+
+mod receive_descriptor;
+use crate::ethernet::receive_descriptor::ReceiveDescriptor;
+
+register_structs! {
+    /// Ethernet: media access control
+/// (MAC)
+    Ethernet_MacRegisters {
+        /// Ethernet MAC configuration
+/// register
+        (0x000 => maccr: ReadWrite<u32, MACCR::Register>),
+        /// Ethernet MAC frame filter
+/// register
+        (0x004 => macffr: ReadWrite<u32, MACFFR::Register>),
+        /// Ethernet MAC hash table high
+/// register
+        (0x008 => machthr: ReadWrite<u32>),
+        /// Ethernet MAC hash table low
+/// register
+        (0x00C => machtlr: ReadWrite<u32>),
+        /// Ethernet MAC MII address
+/// register
+        (0x010 => macmiiar: ReadWrite<u32, MACMIIAR::Register>),
+        /// Ethernet MAC MII data register
+        (0x014 => macmiidr: ReadWrite<u32>),
+        /// Ethernet MAC flow control
+/// register
+        (0x018 => macfcr: ReadWrite<u32, MACFCR::Register>),
+        /// Ethernet MAC VLAN tag register
+        (0x01C => macvlantr: ReadWrite<u32, MACVLANTR::Register>),
+        (0x020 => _reserved0),
+        /// Ethernet MAC PMT control and status
+/// register
+        (0x02C => macpmtcsr: ReadWrite<u32, MACPMTCSR::Register>),
+        (0x030 => _reserved1),
+        /// Ethernet MAC debug register
+        (0x034 => macdbgr: ReadOnly<u32, MACDBGR::Register>),
+        /// Ethernet MAC interrupt status
+/// register
+        (0x038 => macsr: ReadWrite<u32, MACSR::Register>),
+        /// Ethernet MAC interrupt mask
+/// register
+        (0x03C => macimr: ReadWrite<u32, MACIMR::Register>),
+        /// Ethernet MAC address 0 high
+/// register
+        (0x040 => maca0hr: ReadWrite<u32, MACA0HR::Register>),
+        /// Ethernet MAC address 0 low
+/// register
+        (0x044 => maca0lr: ReadWrite<u32>),
+        /// Ethernet MAC address 1 high
+/// register
+        (0x048 => maca1hr: ReadWrite<u32, MACA1HR::Register>),
+        /// Ethernet MAC address1 low
+/// register
+        (0x04C => maca1lr: ReadWrite<u32>),
+        /// Ethernet MAC address 2 high
+/// register
+        (0x050 => maca2hr: ReadWrite<u32, MACA2HR::Register>),
+        /// Ethernet MAC address 2 low
+/// register
+        (0x054 => maca2lr: ReadWrite<u32>),
+        /// Ethernet MAC address 3 high
+/// register
+        (0x058 => maca3hr: ReadWrite<u32, MACA3HR::Register>),
+        /// Ethernet MAC address 3 low
+/// register
+        (0x05C => maca3lr: ReadWrite<u32>),
+        (0x060 => @END),
+    }
+}
+
+register_bitfields![u32,
+MACCR [
+    /// RE
+    RE OFFSET(2) NUMBITS(1) [],
```

Comment:
Perhaps fix the indentation here as well. It's unfortunate that `rustfmt` doesn't really format our custom macros well.